### ISSUE TITLE
Improve sidebar toggle animation

### ISSUE THREAD

**Issue by ajsaraujo** [ajsaraujo/pablo#8]
Sidebar has a little animation for open/closing - it slides in and out. However, `PaletteComponent` doesn't. Add a slide animation on `PaletteComponent` when sidebar is toggled, making it look smoother.

**Comment by vibharaj** [comment]
Hi! Can I work on this?

**Comment by ajsaraujo** [comment]
Yes! Feel free to work on anything you'd like. I don't have much time to review submissions for now, but I'll do it as soon as I can.
